File: app/src/main/java/jp/pikey8706/httpkicksforvlc/ui/main/PageViewModel.java
package jp.pikey8706.httpkicksforvlc.ui.main;

import androidx.arch.core.util.Function;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.Transformations;
import androidx.lifecycle.ViewModel;

public class PageViewModel extends ViewModel {

    private MutableLiveData<String> mChannel = new MutableLiveData<>();
    private LiveData<String> mText = Transformations.map(mChannel, new Function<String, String>() {
        @Override
        public String apply(String channel) {
            return "CH: " + channel;
        }
    });

    public void setChannel(String channel) {
        mChannel.setValue(channel);
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/main/java/jp/pikey8706/httpkicksforvlc/ui/main/PageViewModel.kt
package jp.pikey8706.httpkicksforvlc.ui.main

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.map

class PageViewModel : ViewModel() {
    private val mChannel = MutableLiveData<String>()
    val text = mChannel.map { channel -> "CH: $channel" }

    fun setChannel(channel: String) {
        mChannel.value = channel
    }
}